src/main/java/Wolox/training/controllers/BookController.java

```diff
@@ -45,6 +58,52 @@ public Book findById(@PathVariable int id) {
         return bookRepository.findById(id).orElseThrow(() -> new RuntimeException("The book does not exist"));
     }
 
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public Book notFound() {
+        return null;
+    }
+
+    @ResponseStatus(HttpStatus.OK)
+    @GetMapping
+    public Book findLocalByIsbn(String isbn) {
+        return bookRepository.findByIsbn(isbn).orElseThrow(() -> new NoSuchElementException());
+    }
+
+    @ResponseStatus(HttpStatus.CREATED)
+    @PostMapping
+    public Book findInOnlineLibrary(String isbn) {
+        try {
+            BookDAO bookDAO = onlineLibrary.bookInfo(isbn);
+            Book book = new Book();
+            book.setTitle(bookDAO.getTitle());
+            book.setSubtitle(bookDAO.getSubtitle());
+            book.setPublisher(bookDAO.getPublishers());
+            book.setIsbn(isbn);
+            book.setYear(bookDAO.getPublishDate());
+            book.setAuthor(bookDAO.getAuthors().stream().findFirst().get());
+            book.setImage(bookDAO.getCover());
+            return this.create(book);
+        } catch (RuntimeException e) {
+            throw new NoSuchElementException();
+        }
+    }
+
+    @RequestMapping(value = "/view/isbn/{isbn}", method = { RequestMethod.GET, RequestMethod.POST })
+    public Book findByIsbn(HttpServletResponse response, @PathVariable String isbn) {
+        try {
+            response.setStatus(200);
```

Comment:
Better use the HttpStatus.OK and similars and not the response code directly. 
It's makes the reading easier